lib/ansible/cli/doc.py

```diff
@@ -232,7 +232,7 @@ def _find_all_collection_roles(self, name_filters=None, collection_filter=None):
         b_colldirs = list_collection_dirs(coll_filter=collection_filter)
         for b_path in b_colldirs:
             path = to_text(b_path, errors='surrogate_or_strict')
-            collname = _get_collection_name_from_path(b_path)
+            collname = _get_collection_name_from_path(b_path) or "default_collection"
```

Comment:
Looks like https://github.com/ansible/ansible/issues/84909#issuecomment-2767335761 is the proper solution. And hard-coding an arbitrary string as a fallback breaks stuff. Try going with that instead.